moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp

```diff
@@ -78,6 +78,33 @@ bool RuckigSmoothing::applySmoothing(robot_trajectory::RobotTrajectory& trajecto
   // This lib does not actually work properly when angles wrap around, so we need to unwind the path first
   trajectory.unwind();
 
+  // Create a copy of the desired trajectory so we can revert it as needed.
+  // Remove repeated waypoints with no change in position. Ruckig does not handle this well and there's really no
+  // need to smooth it. Repeated waypoints cause circular motions.
+  robot_trajectory::RobotTrajectory requested_trajectory(trajectory.getRobotModel(), trajectory.getGroup());
+  for (size_t waypoint_idx = 0; waypoint_idx < num_waypoints - 1; ++waypoint_idx)
+  {
+    bool identical_waypoint = checkForIdenticalWaypoints(
+        *trajectory.getWayPointPtr(waypoint_idx), *trajectory.getWayPointPtr(waypoint_idx + 1), trajectory.getGroup());
+    if (identical_waypoint)
+    {
+      continue;
+    }
+    else
+    {
+      requested_trajectory.addSuffixWayPoint(trajectory.getWayPoint(waypoint_idx),
+                                             trajectory.getWayPointDurationFromPrevious(waypoint_idx));
+    }
+  }
+  trajectory = requested_trajectory;
+
+  num_waypoints = trajectory.getWayPointCount();
+  if (num_waypoints < 2)
+  {
+    RCLCPP_ERROR(LOGGER, "Trajectory does not have enough points to smooth with Ruckig");
+    return false;
+  }
+
```

Comment:
Not really related to this PR, but something to keep in mind... The `applySmoothing` function is almost 200 lines long and contains complex logic. That's too long and should be, at some point, split into pieces. I'm thinking about just having helper functions that capture the phases of the smoothing correctly. 

The trajectory processing functions are also all `static` class member functions meaning they really don't have to be in a class but can instead just be simple functions in some suitable namespace.

Reply:
Thanks @jliukkonen. I'm going to close this PR because some refactoring was done and it no longer applies. But I will try to do what you suggest in that PR.